source/aws_signing.c

```diff
@@ -2399,38 +2408,46 @@ int aws_verify_sigv4a_signing(
         AWS_BYTE_CURSOR_PRI(ecc_key_pub_x),
         AWS_BYTE_CURSOR_PRI(ecc_key_pub_y));
 
-    struct aws_ecc_key_pair *verification_key =
-        aws_ecc_key_new_from_hex_coordinates(allocator, AWS_CAL_ECDSA_P256, ecc_key_pub_x, ecc_key_pub_y);
-    if (verification_key == NULL) {
-        goto done;
-    }
-
-    if (aws_credentials_get_ecc_key_pair(signing_state->config.credentials) == NULL) {
-        struct aws_credentials *ecc_credentials =
-            aws_credentials_new_ecc_from_aws_credentials(allocator, signing_state->config.credentials);
-        aws_credentials_release(signing_state->config.credentials);
-        signing_state->config.credentials = ecc_credentials;
-        if (signing_state->config.credentials == NULL) {
-            goto done;
-        }
-    }
+    struct aws_ecc_key_pair *verification_key = NULL;
```

Comment:
Why not to initialize at the time of declaration? I don't see any use of verification_key until the initialization line in 2430

Reply:
because otherwise you get goto issues, and I don't want to touch ECC stuff until after everything else is validated